request/executor.go

```diff
@@ -7,44 +7,85 @@ import (
 )
 
 // ExecuteRequest executes an HTTP request based on the specified options.
-func ExecuteRequest(request Request, profileNames []string, variables map[string]string) (*ExecutedRequestResponse, error) {
-	httpRequest, configuredRequest, httpRequestErr := BuildRequest(request, profileNames, variables)
-	if httpRequestErr != nil {
-		return nil, httpRequestErr
-	}
+func ExecuteRequest(request Request, profileNames []string, variables map[string]string) ([]ExecutedRequestResponse, error) {
+	toExecute := make([]Request, 1)
+	toExecute[0] = request
 
-	client := &http.Client{}
-	httpResponse, httpResponseErr := client.Do(httpRequest)
-	if httpResponseErr != nil {
-		return nil, httpResponseErr
+	client := &http.Client{
+		// Do not auto-follow redirects
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
 	}
 
-	bodyBytes, readErr := ioutil.ReadAll(httpResponse.Body)
+	result := make([]ExecutedRequestResponse, 0)
 
-	if readErr != nil {
-		return nil, readErr
-	}
+	for {
+		executing := toExecute[0]
+		toExecute = toExecute[1:]
 
-	body := ""
-	if len(bodyBytes) != 0 {
-		body = string(bodyBytes)
-	}
+		httpRequest, configuredRequest, httpRequestErr := BuildRequest(executing, profileNames, variables)
+		if httpRequestErr != nil {
+			return nil, httpRequestErr
+		}
 
-	headers := make(map[string][]string)
-	for k, vs := range httpResponse.Header {
-		headers[k] = append(headers[k], vs...)
-	}
+		httpResponse, httpResponseErr := client.Do(httpRequest)
+		if httpResponseErr != nil {
+			return nil, httpResponseErr
+		}
+
+		bodyBytes, readErr := ioutil.ReadAll(httpResponse.Body)
+
+		if readErr != nil {
+			return nil, readErr
+		}
+
+		body := ""
+		if len(bodyBytes) != 0 {
```

Comment:
I could definitely be missing something with golang but why not
```golang
body := string(bodyBytes)
```
![image](https://user-images.githubusercontent.com/8136030/34802100-c2d19e22-f639-11e7-8be6-4e479348bb4f.png)

Reply:
Good catch, I don't even need that variable.